tests/datavolume_test.go

```diff
@@ -811,6 +813,50 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
 		})
 	})
 
+	Describe("Registry import with missing configmap", func() {
+		const cmName = "cert-registry-cm"
+
+		It("Import POD should remain pending until CM exists", func() {
+			var pvc *corev1.PersistentVolumeClaim
+
+			dataVolumeDef := utils.NewDataVolumeWithRegistryImport("missing-cm-registry-dv", "1Gi", tinyCoreIsoRegistryURL)
+			dataVolumeDef.Spec.Source.Registry.CertConfigMap = cmName
+			dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolumeDef)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("verifying pvc was created")
+			Eventually(func() bool {
+				// TODO: fix this to use the mechanism to find the correct PVC once we decouple the DV and PVC names
+				pvc, _ = f.K8sClient.CoreV1().PersistentVolumeClaims(dataVolume.Namespace).Get(dataVolume.Name, metav1.GetOptions{})
+				return pvc != nil && pvc.Name != ""
+			}, timeout, pollingInterval).Should(BeTrue())
+
+			By("Verifying the POD remains pending for 30 seconds")
+			podName := naming.GetResourceName(common.ImporterPodName, pvc.Name)
+			for i := 0; i < 30; i++ {
+				pod, err := f.K8sClient.CoreV1().Pods(f.Namespace.Name).Get(podName, metav1.GetOptions{})
+				if err == nil {
+					// Found the pod
+					Expect(pod.Status.Phase).To(Equal(corev1.PodPending))
+					Expect(pod.Status.ContainerStatuses[0].State.Waiting.Reason).To(Equal("ContainerCreating"))
+					fmt.Fprintf(GinkgoWriter, "INFO: pod found, pending, container creating: %s\n", podName)
+				} else if k8serrors.IsNotFound(err) {
+					fmt.Fprintf(GinkgoWriter, "INFO: pod not found: %s\n", podName)
+				} else {
+					Expect(err).ToNot(HaveOccurred())
+				}
+				// Sleep one second and try again.
+				time.Sleep(time.Second)
+			}
+
+			By("Creating the config map")
+			_, err = utils.CopyRegistryCertConfigMapDestName(f.K8sClient, f.Namespace.Name, f.CdiInstallNs, cmName)
+			Expect(err).To(BeNil())
+
+			By(fmt.Sprintf("waiting for datavolume to match phase %s", string(cdiv1.Succeeded)))
+			utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, dataVolume.Name)
```

Comment:
Missing check of return value?

also: I assume the cleanup is handled by the namespace choice, is that correct?

Reply:
Yeah part of the framework is deleting the namespace at the end, which will in turn delete the data volume/pvc/pods that are in that namespace. Let me add the check.